src/main.rs

```diff
@@ -122,59 +123,74 @@ fn main() -> anyhow::Result<()> {
                 .deps(false)
                 .silence(!config.is_verbose());
 
-            let rustdoc_paths = if let Some(current_rustdoc_path) = args.current_rustdoc.as_deref()
+            let all_outcomes: Vec<anyhow::Result<bool>> = if let Some(current_rustdoc_path) =
+                args.current_rustdoc.as_deref()
             {
+                let current_crate = load_rustdoc(current_rustdoc_path)?;
+
                 let name = "<unknown>";
                 let version = None;
-                vec![(
-                    name.to_owned(),
-                    loader.load_rustdoc(&mut config, &rustdoc_cmd, name, version)?,
-                    current_rustdoc_path.to_owned(),
-                )]
+                let baseline_path =
+                    loader.load_rustdoc(&mut config, &rustdoc_cmd, name, version)?;
+                let baseline_crate = load_rustdoc(&baseline_path)?;
```

Comment:
Would it be possible to extract a function that generates the `(current_crate, baseline_crate)` tuple?

Asking because this code here can also be affected by the same JSON file overwriting bug as the code below near L167-178. That other code block has the big comment, which is great, but this part here doesn't. Instead of duplicating the comment, perhaps we can extract the logic into a function and put the big comment block on that function?